service/common/src/main/java/org/apache/polaris/service/admin/PolarisServiceImpl.java

```diff
@@ -93,26 +94,32 @@ public PolarisServiceImpl(
     this.polarisAuthorizer = polarisAuthorizer;
   }
 
-  private PolarisAdminService newAdminService(SecurityContext securityContext) {
-    CallContext callContext = CallContext.getCurrentContext();
+  private PolarisAdminService newAdminService(
+      RealmContext realmContext, SecurityContext securityContext) {
     AuthenticatedPolarisPrincipal authenticatedPrincipal =
         (AuthenticatedPolarisPrincipal) securityContext.getUserPrincipal();
     if (authenticatedPrincipal == null) {
       throw new NotAuthorizedException("Failed to find authenticatedPrincipal in SecurityContext");
     }
 
     PolarisEntityManager entityManager =
-        entityManagerFactory.getOrCreateEntityManager(callContext.getRealmContext());
+        entityManagerFactory.getOrCreateEntityManager(realmContext);
```

Comment:
Can `RealmContext` be a `@RequestScoped` injectable (constructor) property as opposed to an explicit method parameter?

Reply:
No, for 2 reasons: 1) DW does not "understand" request-scoped REST resources, and 2) Quarkus will need the parameter anyway, to add the realm id to the generated metric tags. IOW, later on the parameter will be annotated with `@MeterTag` as follows:


```java
@Context 
@MeterTag(key="realm_id",expression="realmIdentifier") 
RealmContext realmContext,
```

Reply:
I see :+1:

Reply:
Hypothetically: is it possible to manually produce a `@RequestScoped` bean for the realm context data in the generated API code?